src/core/internal/hash.d

```diff
@@ -423,6 +423,31 @@ size_t hashOf(T)(scope const T val, size_t seed = 0) if (!is(T == enum) && __tra
     }
 }
 
+size_t hashOf(T)(scope const auto ref T val, size_t seed = 0) @safe @nogc nothrow pure
+if (is(T == __vector) && !is(T == enum))
+{
+    static if (__traits(isFloating, T) && (floatCoalesceZeroes || floatCoalesceNaNs))
+    {
+        if (__ctfe)
+        {
+            // Workaround for CTFE bug.
+            size_t result = seed;
+            alias E = typeof(val[0]);
+            foreach (i; 0 .. T.sizeof / E.sizeof)
+                result = hashOf(val[i], result);
+            return result;
+        }
+        size_t result = seed;
+        foreach (x; val.array)
+            result = hashOf(x, result);
+        return result;
```

Comment:
Why not keep only the CTFE version?

Reply:
It seemed to produce better code with DMD. It looks like LDC2 compiles them both the same though.